plugins/modules/opkg.py

```diff
@@ -102,16 +109,36 @@ def update_package_db(module, opkg_path):
         module.fail_json(msg="could not update package db")
 
 
-def query_package(module, opkg_path, name, state="present"):
+def query_package(module, opkg_path, name, version=None, state="present"):
     """ Returns whether a package is installed or not. """
 
     if state == "present":
-
-        rc, out, err = module.run_command("%s list-installed | grep -q \"^%s \"" % (shlex_quote(opkg_path), shlex_quote(name)), use_unsafe_shell=True)
-        if rc == 0:
-            return True
-
-        return False
+        rc, out, err = module.run_command("%s list-installed %s" % (opkg_path, name))
```

Comment:
Better use
```suggestion
        rc, out, err = module.run_command([opkg_path, "list-installed", name])
```
Otherwise you need to use `shlex_quote` to avoid the shell parsing this wrongly.

Reply:
Changed in [beaf2a1](https://github.com/ansible-collections/community.general/pull/5688/commits/beaf2a186bed1eda88a8f7c1243dc2be4aba6290)